### PR TITLE
prover: make sure the block sizes are sorted

### PR DESCRIPTION
**Pull request by yutianwu**
### Description

This pr will make sure the block sizes config in the prover are sorted.

/update-integration-keyfile

### Changes

Notable changes:
* make sure the block sizes are sorted

### CODE REVIEW

File: service/prover/prover/prover.go [hunk 1]
```diff
@@ -42,6 +42,16 @@ func WithRedis(redisType string, redisPass string) redis.Option {
 		p.Pass = redisPass
 	}
 }
+
+func IsBlockSizesSorted(blockSizes []int) bool {
+	for i := 1; i < len(blockSizes); i++ {
+		if blockSizes[i] < blockSizes[i-1] {
```
Comment (lightning-li) [inline service/prover/prover/prover.go:48]: Should use <= rather than <.

Comment (KeefeL) [inline service/prover/prover/prover.go:48]: < -> <= ?

**Review by lightning-li — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -84,7 +84,7 @@ require (
 )
 
 require (
-	github.com/bnb-chain/zkbnb-crypto v0.0.6-0.20220907124239-7cdc9ddd1728
+	github.com/bnb-chain/zkbnb-crypto v0.0.4-0.20220908063720-b916cd9542fd
```
Comment (guagualvcha) [inline go.mod:87]: Why this degraded?

File: service/prover/prover/prover.go [hunk 3]
```diff
@@ -42,6 +42,16 @@ func WithRedis(redisType string, redisPass string) redis.Option {
 		p.Pass = redisPass
 	}
 }
+
+func IsBlockSizesSorted(blockSizes []int) bool {
+	for i := 1; i < len(blockSizes); i++ {
```
Comment (guagualvcha) [inline service/prover/prover/prover.go:47]: `i` start from 1 may case index out of range.

Comment (yutianwu) [reply service/prover/prover/prover.go:47]: in that case, 1 should larger than `len(blockSizes)`

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**